compiler/rustc_ast_lowering/src/index.rs

```diff
@@ -219,6 +225,11 @@ impl<'a, 'hir> Visitor<'hir> for NodeCollector<'a, 'hir> {
 
     fn visit_expr(&mut self, expr: &'hir Expr<'hir>) {
         self.insert(expr.span, expr.hir_id, Node::Expr(expr));
+        if let ExprKind::Struct(_, fields, _) = expr.kind {
+            for field in fields {
+                self.insert(field.span, field.hir_id, Node::ExprField(field));
```

Comment:
Can this be moved into visit_expr_field?

Reply:
Moved. That changed the parent relationship, which I think is correct, but required modifications on various things looking at the parent.